sdk/resourcemanager/Azure.ResourceManager.Core/README.md

```diff
@@ -0,0 +1,169 @@
+// ResourceGroup is a [Resource] object from above
+ResourceGroup resourceGroup = await armClient.DefaultSubscription.GetResourceGroups().GetAsync("myRgName");
+
+// Next we get the container for the virtual machines
+// vmContainer is a [Resource]Container object from above
+VirtualMachineContainer vmContainer = resourceGroup.GetVirtualMachines();
+
+// Next we loop over all vms in the container
+// Each vm is a [Resource] object from above
+await foreach(VirtualMachine vm in vmContainer.ListAsync())
+{
+    // We access the [Resource]Data properties from vm.Data
+    if(!vm.Data.Tags.ContainsKey("owner"))
+    {
+        // We can also access all [Resource]Operations from vm since it is already scoped for us
+        await vm.StartAddTag("owner", GetOwner()).WaitForCompletionAsync();
+    }
+}
+ ```
+
+### Create a resource group
+```csharp
+// First, initialize the ArmClient and get the default subscription
+var armClient = new ArmClient(new DefaultAzureCredential());
+Subscription subscription = armClient.DefaultSubscription;
+// Now we get a ResourceGroup container for that subscription
+ResourceGroupContainer rgContainer = subscription.GetResourceGroups();
+
+// With the container, we can create a new resource group with an specific name
+string rgName = "myRgName";
+ResourceGroup resourceGroup = await rgContainer.CreateAsync(rgName);
+```
+
+### List all resource groups
+```csharp
+// First, initialize the ArmClient and get the default subscription
+var armClient = new ArmClient(new DefaultAzureCredential());
+Subscription subscription = armClient.DefaultSubscription;
+
+// Now we get a ResourceGroup container for that subscription
+ResourceGroupContainer rgContainer = subscription.GetResourceGroups();
+
+// With ListAsync(), we can get a list of the resources in the container
+AsyncPageable<ResourceGroup> response = rgContainer.ListAsync();
+await foreach (ResourceGroup rg in response)
+{
+    Console.WriteLine(rg.Data.Name);
+}
+```
+
+For more detailed examples, take a look at [samples](https://github.com/Azure/azure-sdk-for-net/tree/feature/mgmt-track2/sdk/resourcemanager/Azure.ResourceManager.Core/samples) we have available.
+
+## Troubleshooting
+
+-   If you find a bug or have a suggestion, file an issue via [GitHub issues](https://github.com/Azure/azure-sdk-for-net/issues) and make sure you add the "Preview" label to the issue.
+-   If you need help, check [previous
+    questions](https://stackoverflow.com/questions/tagged/azure+.net)
+    or ask new ones on StackOverflow using azure and .NET tags.
+-   If having trouble with authentication, go to [DefaultAzureCredential documentation](https://docs.microsoft.com/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet).
+## Next steps
+### More sample code
+
+- [Managing Resource Groups](https://github.com/Azure/azure-sdk-for-net/blob/feature/mgmt-track2/sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample2_ManagingResourceGroups.md)
```

Comment:
You will probably want to update these links to point to main once this goes into main.